src/main/java/com/fastcampus/projectboard/domain/Article.java

```diff
@@ -1,14 +1,69 @@
 package com.fastcampus.projectboard.domain;
 
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import org.springframework.data.annotation.CreatedBy;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedBy;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.*;
 import java.time.LocalDateTime;
+import java.util.Objects;
 
+@Getter
+@ToString
+@Table(indexes = {
+        @Index(columnList = "title"),
+        @Index(columnList = "hashtag"),
+        @Index(columnList = "createdAt"),
+        @Index(columnList = "createdBy")
+})
+@EntityListeners(AuditingEntityListener.class)
+@Entity
 public class Article {
-    private String title;
-    private String content;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false) private String title;
+    @Column(nullable = false, length = 10000) private String content;
     private String hashtag;
 
-    private LocalDateTime createdAt;
-    private String createdBy;
-    private LocalDateTime modifiedAt;
-    private String ModifiedBy;
+    @CreatedDate @Column(nullable = false) private LocalDateTime createdAt;
+    @CreatedBy @Column(nullable = false,length = 100) private String createdBy;
+    @LastModifiedDate @Column(nullable = false) private LocalDateTime modifiedAt;
+    @LastModifiedBy @Column(nullable = false,length = 100) private String modifiedBy;
+
+    protected Article() {}
+
+    public Article(String title, String content, String hashtag) {
+        this.title = title;
+        this.content = content;
+        this.hashtag = hashtag;
+    }
+
+    public static Article of(String title, String content, String hashtag) {
+        return new Article(title,content, hashtag);
+    }
```

Comment:
이 생성자는 정적 메소드 `of()`와 연결되어 있습니다. 생성자의 접근자를 막아보는 것은 어떨까요?
그렇게 하면 사용자는 엔티티를 생성할 때마다 `of()` 사용이 강제되지 않을까요?
정답이 있는 것이 아니니 생각해 보시고 의견 주세요.

```suggestion
    private Article(String title, String content, String hashtag) {
        this.title = title;
        this.content = content;
        this.hashtag = hashtag;
    }

    public static Article of(String title, String content, String hashtag) {
        return new Article(title,content, hashtag);
    }
```

Reply:
넵 맞습니다. 
정적 팩토리 메서드의 경우 객체를 생성할 때 보다 이해하기 쉽고 유연하게 만들기 위해 사용됩니다.

of 함수가 생성자를 호출하는데 생성자가 public으로 열려있다면 정적 팩토리 메서드를 만드는 이유가 없겠네요
감사합니다! 수정해놓겠습니다!

Reply:
제 생각은 달라요. 수업 시간에 살짝 설명드린 것처럼, 팩토리 메소드는 유연하게 만들기 위해 쓰는 것이 아니라 반대로 딱딱하게 만들기 위해 쓰는 것입니다.
생성자가 오픈되어있지 않고, 사용자가 오로지 팩토리 메소드만을 이용해 인스턴스를 생성할 수 있다면, 사용자는 설계자가 의도한 것에 맞춰서
팩토리 메소드에 데이터를 반드시 전달해야지만 인스턴스를 만들 수 있습니다. 이는 제약이고, 사용법을 설계를 통해 사용자에게 보여준 것입니다.
자유롭게 만들 수 없으니 딱딱하다고 봐야겠죠.

그리고 생성자가 퍼블릭으로 열려있다고 하더라도 팩토리 메소드를 갖는 것이 무의미해지는 것은 아닙니다. 단지 사용자에게 여러 선택지를 주는 것이 됩니다.
그것이 설계자의 의도나 설계 목적에 부합하는지를 판단하여 결정하면 되겠습니다.